sdk/src/main/java/io/opentelemetry/sdk/metrics/LongObserverSdk.java

```diff
@@ -34,8 +43,31 @@
   }
 
   @Override
-  public void setCallback(Callback<LongObserver.ResultLongObserver> metricUpdater) {
-    throw new UnsupportedOperationException("to be implemented");
+  List<MetricData> collectAll() {
+    Callback<ResultLongObserver> currentMetricUpdater = metricUpdater;
+    if (currentMetricUpdater == null) {
+      return Collections.emptyList();
+    }
+    final ActiveBatcher activeBatcher = getActiveBatcher();
+    currentMetricUpdater.update(
+        new ResultLongObserver() {
```

Comment:
Probably a minor issue, since this is only called once per collection cycle, but I'd prefer to see this be a concrete, rather than anonymous class.

Reply:
Done.